web/src/main/react-dashboard/src/components/TestRunView.js

```diff
@@ -49,563 +42,570 @@ import {FAIL, SUCCESS, ERROR, PENDING, RUNNING, HTTP_UNAUTHORIZED, LOGIN_URI, TE
-                          style={{
-                            verticalAlign: "middle",
-                            height: "50px",
-                            width: "50px"
-                          }} />
-                      </td>
-                      {subHeader}
-                    </tr>
-                  </tbody>
-                </table>
-              </Subheader>;
-            case PENDING:
-            case RUNNING:
-            default:
-              return <Subheader style={{
-                fontSize: '20px',
-                backgroundColor: "#d7d9ff"
-              }}>
-                <table>
-                  <tbody>
-                    <tr>
-                      <td style={{ padding: 5 }}>
-                        <CircularProgress size={80} thickness={8} />
-                      </td>
-                      {subHeader}
-                    </tr>
-                  </tbody>
-                </table>
-              </Subheader>;
-          }
-        })()}
-        <Card style={{ padding: 20 }}>
-          <CardMedia>
-            {/*TestGrid generated contents*/}
-            <List>
-              <ListItem disabled={true}
-                leftAvatar={
-                  <Avatar
-                    src={require('../log.png')}
-                    size={50}
-                    style={{
-                      borderRadius: 0,
-                      backgroundColor: "#ffffff"
-                    }} />
-                }>
-                <FlatButton label="Download Test Run Log"
-                  onClick={() => (fetch(logAllContentUrl, {
-                    method: "GET",
-                    credentials: 'same-origin',
-                    headers: {
-                      'Accept': 'application/json'
-                    }
-                  })
-                  .then(this.handleError)
-                  .then(response => {
-                    this.setState({
-                      showLogDownloadErrorDialog: !response.ok
-                    });
-                    return response.json();
-                  }).then(data => {
-                    if (!this.state.showLogDownloadErrorDialog) {
-                      Download(data.inputStreamContent, "test-run.log", "plain/text");
+        return (
+            <div>
+                {/*Sub header*/}
+                {this.state && this.state.currentInfra && (() => {
+                    const subHeader = (<td style={{padding: 5}}>
+                        <i> {this.state.currentInfra.relatedProduct}/
+                            {this.state.currentInfra.relatedDeplymentPattern} /
+                            {this.state.currentInfra.infraParameters}</i>
+                    </td>);
+                    switch (this.state.currentInfra.testPlanStatus) {
+                        case FAIL:
+                            return <Subheader style={{
+                                fontSize: '20px',
+                                backgroundColor: "#ffd6d3"
```

Comment:
Is it advisable set css styling info within the js files like this? What's the best practice for better maintainability of the code?

Reply:
AFAIK inline CSS is not a good practice. Let's raise a ticket and handle these kind of things. WDYT?

Reply:
+1